agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -3249,23 +3249,25 @@ bool slave_thread::slave_fsm(bool &call_slave_select)
         break;
     }
     case STATE_CONNECT_TO_BACKHAUL_MANAGER: {
-        if (backhaul_manager_socket) {
-            remove_socket(backhaul_manager_socket);
-            delete backhaul_manager_socket;
-            backhaul_manager_socket = nullptr;
-        }
+        if (backhaul_manager_socket == nullptr) {
+            LOG(DEBUG) << "create backhaul_manager_socket";
         backhaul_manager_socket = new SocketClient(backhaul_manager_uds);
         std::string err         = backhaul_manager_socket->getError();
         if (!err.empty()) {
-            delete backhaul_manager_socket;
-            backhaul_manager_socket = nullptr;
             LOG(ERROR) << "backhaul_manager_socket: " << err;
+                backhaul_manager_stop();
             platform_notify_error(BPL_ERR_SLAVE_CONNECTING_TO_BACKHAUL_MANAGER,
                                   "iface=" + config.backhaul_wireless_iface);
             stop_on_failure_attempts--;
             slave_reset();
+                break;
         } else {
             add_socket(backhaul_manager_socket);
+            }
```

Comment:
Indentation is wrong in "agent: Backhaul manager socket reuse if exist" commit, fixed later. (nitpick)